service/common/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -191,6 +192,7 @@ static int mapExceptionToResponseCode(RuntimeException rex) {
       case IllegalArgumentException e -> Status.BAD_REQUEST.getStatusCode();
       case UnsupportedOperationException e -> Status.NOT_ACCEPTABLE.getStatusCode();
       case WebApplicationException e -> e.getResponse().getStatus();
+      case PersistenceException e -> Status.UNAUTHORIZED.getStatusCode();
```

Comment:
This may not be enough to fix the issue, as line 105 to line 115 still leak the detailed database information.

Another concern is that, the error log has gone after the change as we changed the `Status code` to 401. I think we still need to log an error in the server side.
```
    if (responseCode == Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()) {
      LOGGER.error("Unhandled exception returning INTERNAL_SERVER_ERROR", runtimeException);
    }
```

Reply:
Thank you for your feedback 🙏 
This makes total sense. Instead of extending the existing mapper, I've created a new one where I set a message from PersistenceException but shouldn't include the information like `Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.4.v202407190748...`

I've checked existing mappers and I see that we use a level INFO for every known exception I've chosen an INFO level as well instead of an ERROR level